.teamcity/Gradle_Check/configurations/PerformanceTest.kt

```diff
@@ -25,25 +28,44 @@ class PerformanceTest(model: CIBuildModel, type: PerformanceTestType, stage: Sta
         }
     }
 
+    requirements {
+        // We observed some relocatability issues with its inputs and it is a long running job
+        doesNotContain("teamcity.agent.name", "ec2")
+    }
+
     params {
         param("performance.baselines", type.defaultBaselines)
     }
 
-    steps {
+    fun BuildSteps.runner(runnerName: String, runnerTasks: String, extraParameters: String = "", runnerExecutionMode: BuildStep.ExecutionMode = ExecutionMode.DEFAULT) {
         gradleWrapper {
-            name = "GRADLE_RUNNER"
+            name = runnerName
             tasks = ""
-            gradleParams = (
-                    buildToolGradleParameters(isContinue = false)
-                        + performanceTestCommandLine(task = "distributed${type.taskId}s", baselines = "%performance.baselines%", extraParameters = type.extraParameters)
-                        + distributedPerformanceTestParameters(IndividualPerformanceScenarioWorkers(model).id.toString())
-                        + listOf(buildScanTag("PerformanceTest"))
-                        + model.parentBuildCache.gradleParameters(Os.linux)
+            executionMode = runnerExecutionMode
+            gradleParams = (performanceTestCommandLine(task = runnerTasks, baselines = "%performance.baselines%", extraParameters = type.extraParameters)
+                    + buildToolGradleParameters(isContinue = false)
+                    + distributedPerformanceTestParameters(IndividualPerformanceScenarioWorkers(model).id.toString())
+                    + listOf(buildScanTag("PerformanceTest"))
+                    + model.parentBuildCache.gradleParameters(Os.linux)
+                    + extraParameters
                     ).joinToString(separator = " ")
         }
+    }
+
+    steps {
+        runner("GRADLE_RUNNER", "clean distributed${type.taskId}s")
         checkCleanM2()
-        tagBuild(model, true)
+        if (type.hasRerunner) {
+            val rerunnerParameters = listOf(
+                    "-PteamCityBuildId=%teamcity.build.id%",
+                    "-PonlyPreviousFailedTestClasses=true",
+                    "-PgithubToken=%github.ci.oauth.token%")
+            runner("GRADLE_RERUNNER", "tagBuild distributed${type.taskId}s", rerunnerParameters.joinToString(" "), ExecutionMode.RUN_ON_FAILURE)
```

Comment:
I just wondered where the tagging with `RERUN_TESTS` happens and realized it happens in `BuildScanPlugin`. I guess we should move the tagging to the plugin in `ci-health` as well. WDYT?

Reply:
Tagging with `RERUN_TESTS` has two dimensions: it's both "from rerun" and "from scan". I think currently we put everything related to build scan in `BuildScanPlugin` (maybe because `BuildScanPlugin` has too many things to set up and configure? e.g. `BuildScanExtension`)

I would keep it in `BuildScanPlugin`, just because moving it to ci-health is a little complicated.

Reply:
Why is it complicated? It just means that, when `onlyPreviousFailedTestClasses` is passed, then the build scan tag should be added. There is even the option to add it via our Kotlin TC DSL scripts via a system property (scan tags can be added by setting a system property).

Reply:
By complicated I mean ci-health project needs to depend on build scan jar and add tags via `BuildScanExtension`. 

> scan tags can be added by setting a system property

Good to know, I'd prefer to add the tag via Kotlin DSL so that we can have these stuff in one place.